src/test/java/seedu/partyplanet/model/person/BirthdayTest.java

```diff
@@ -0,0 +1,121 @@
+    public void isValidBirthday_invalidBirthdays() {
+        // null birthdays
+        assertThrows(NullPointerException.class, () -> Birthday.isValidBirthday(null));
+
+        // invalid birthdays
+        assertFalse(Birthday.isValidBirthday("")); // empty string
+        assertFalse(Birthday.isValidBirthday(" ")); // spaces only
+        assertFalse(Birthday.isValidBirthday("91")); // none match
+        assertFalse(Birthday.isValidBirthday("2021 Feb 20")); // unsupported format
+        assertFalse(Birthday.isValidBirthday("20Feb2021")); // unsupported format
+    }
+
+    @Test
+    public void isValidBirthday_validWithYears() {
+        String[] validInputs = new String[]{
+            "2021-02-03",
+            "3.2.2021",
+            "3/2/2021",
+            "3 Feb 2021",
+            "3 February 2021",
+            "Feb 3 2021",
+            "February 3 2021",
+        };
+        for (String validInput: validInputs) {
+            assertTrue(Birthday.isValidBirthday(validInput));
+        }
+        assertEquals(1, Arrays.stream(validInputs)
+                .map(Birthday::parseBirthday)
+                .map(x -> (LocalDate) x)
+                .distinct()
+                .count());
+    }
+
+    @Test
+    public void isValidBirthday_validWithoutYears() {
+        String[] validInputs = new String[]{
+            "--02-03",
+            "3/2",
+            "3 Feb",
+            "3 February",
+            "Feb 3",
+            "February 3",
+        };
+        for (String validInput: validInputs) {
+            assertTrue(Birthday.isValidBirthday(validInput));
+        }
+        assertEquals(1, Arrays.stream(validInputs)
+                .map(Birthday::parseBirthday)
+                .map(x -> (MonthDay) x)
+                .distinct()
+                .count());
+    }
+
+    @Test
+    public void isValidBirthdayDate() {
+        LocalDate referenceDate = LocalDate.of(2021, 6, 6);
+
+        // MonthDay is always valid
+        assertTrue(Birthday.isValidBirthdayDate("--02-03", referenceDate));
+        assertTrue(Birthday.isValidBirthdayDate("--06-06", referenceDate));
+        assertTrue(Birthday.isValidBirthdayDate("--12-12", referenceDate));
+
+        // LocalDate must be before reference
+        assertTrue(Birthday.isValidBirthdayDate("2021-06-05", referenceDate));
+        assertTrue(Birthday.isValidBirthdayDate("2021-06-06", referenceDate));
+        assertFalse(Birthday.isValidBirthdayDate("2021-06-07", referenceDate));
+    }
+
+    @Test
+    public void compareTo() {
+        // Same year
+        Birthday localDate0606 = new Birthday("2020-06-06");
+        Birthday monthDay0606 = new Birthday("--06-06");
+        Birthday monthDay0607 = new Birthday("--06-07");
+        Birthday localDate0608 = new Birthday("2020-06-08");
+
+        // Different year
+        Birthday localDatePrevYear = new Birthday("2019-08-13");
+
+        // Same date
+        assertEquals(0, localDate0606.compareTo(localDate0606));
```

Comment:
I detect that this code is problematic. According to the [Correctness (CORRECTNESS)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#correctness-correctness), [SA: Self comparison of value with itself (SA_LOCAL_SELF_COMPARISON)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#sa-self-comparison-of-value-with-itself-sa-local-self-comparison).
This method compares a local variable with itself, and may indicate a typo or a logic error. Make sure that you are comparing the right things.